lang/de/availability_shibboleth2fa.php

```diff
@@ -27,7 +27,7 @@
 $string['description'] = 'Erfordere eine Zwei-Faktor-Authentifizierung.';
 $string['eventuser2faloggedin'] = 'Nutzer/in hat sich mit einem zweiten Faktor authentifiziert';
 $string['fulltitle'] = 'Zwei-Faktor-Authentifizierung';
-$string['login_failed'] = 'Der Authentifizierungs-Vorgang ist fehlgeschlagen. Bitte kontaktieren Sie Ihren Server-Administrator oder versuchen Sie es später erneut.';
+$string['login_failed'] = 'Der Authentifizierungsvorgang ist fehlgeschlagen. Bitte kontaktieren Sie Ihren Administrator oder versuchen Sie es später erneut.';
```

Comment:
Interesting that there seem to be no checks in the CI pipeline for this. I know that Moodle has these (questionable) rules for variable names, but I could not find any section in their docs prescribing the same naming convention for language string keys. Specifically this is _not_ mentioned [here](https://moodledev.io/general/development/policies/codingstyle#language-strings). @larsbonczek can you point me to where this is written? Maybe they removed it or am I just blind?

I know we used to stick to this naming convention before, but if this is not written anywhere (anymore?) and not enforced by the CI tool, I would rather use _more_ underscores in the language string keys, not less.

Also, it seems even Moodle core does not stick to this naming style. See for example [here](https://github.com/moodle/moodle/blob/main/mod/forum/lang/en/forum.php) or [here](https://github.com/moodle/moodle/blob/main/lang/en/availability.php).

Reply:
I cannot actually find any place in the Moodle docs where this is stated explicitly, but I would definitely say that lowercase string keys (without underscores) are far more prevalent in the Moodle code base than string keys with underscores.

One exception is string keys ending in `_help`, `_link` and `_desc` (see [here](https://docs.moodle.org/dev/String_API#Help_strings)) and string keys starting with `cachedef_` (see [here](https://moodledev.io/docs/4.5/apis/subsystems/muc#creating-a-definition)), as in the [first example](https://github.com/moodle/moodle/blob/main/mod/forum/lang/en/forum.php) you gave. As for the [second example](https://github.com/moodle/moodle/blob/main/lang/en/availability.php), I am sure that for any Moodle coding style convention there is, you could find some Moodle Core code which does not follow it. In fact, that language file also seems to use [UI concatenation](https://github.com/moodle/moodle/blob/main/lang/en/availability.php#L56-L59) which is explicitly forbidden by the [Coding style](https://moodledev.io/general/development/policies/codingstyle#structure). So I wouldn't count on this code's reliability when it comes to style conventions.

I don't think we necessarily need to address this here, but we should decide on one version sooner or later.

Reply:
I agree we should decide on our own style, if this is not set by Moodle and I agree we don't need to do this now.

And since these string keys were not changed by me in this PR anyway (I just sorted them), can we consider this discussion resolved?